test/e2e/cron/cron_test.go

```diff
@@ -0,0 +1,147 @@
+package cron
+
+import (
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo/test/e2e/fixtures"
+	"github.com/argoproj/argo/workflow/common"
+	"github.com/argoproj/pkg/humanize"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"testing"
+	"time"
+)
+
+type CronSuite struct {
+	fixtures.E2ESuite
+}
+
+func (s *CronSuite) TestBasic() {
+	s.T().Parallel()
+	s.Given().
+		CronWorkflow("@testdata/basic.yaml").
+		When().
+		CreateCronWorkflow().
+		Wait(1 * time.Minute).
+		Then().
+		ExpectCron(func(t *testing.T, cronWf *wfv1.CronWorkflowStatus) {
+			assert.True(t, cronWf.LastScheduledTime.Time.After(time.Now().Add(-1*time.Minute)))
+		})
+}
+
+func (s *CronSuite) TestBasicForbid() {
+	s.T().Parallel()
+	s.Given().
+		CronWorkflow("@testdata/basic-forbid.yaml").
+		When().
+		CreateCronWorkflow().
+		Wait(2 * time.Minute).
```

Comment:
anyway to reduce this time to <30s?

Reply:
Unfortunately, I don't think so. I need at least two cron cycles for this test, the least significant of which is a minute. I could mock the cron frame work to run -- say -- every 15 seconds, but since we are dealing with k8s objects I worry that the short time might cause these tests to be flakey. To mediate the effects of this I decided to run these particular tests in parallel, so the entire cron test suite should take at most 3 minutes.

Let me know what you think.